Test cases for GRAPE (#37)
genxnetwork/grape: test_cases.md

> # List of testing cases before release

Reviewer: ```suggestion
# List of test cases
```

### Accuracy check via simulation: IBIS workflow

**Description:**

We are checking accuracy of IBIS workflow with `simulate` command. 

> It will take roughly an hour. 

Reviewer: ```suggestion
It takes roughly an hour. 
```

**Command:**

```

> docker run --rm --privileged -it -v /media/data1/relatives:/media -v /etc/localtime:/etc/localtime:ro genx_relatives:latest launcher.py simulate --ref-directory /media/big_ref --cores 8 --directory /media/runs/sim-ibis-small --flow ibis --assembly hg37 --real-run

Reviewer: May it's better to write something about `/media/data1/relative`, `/media/big_ref` directories. What should we store there? How to get reference data?

```

**Desired result:**

> Accuracy of almost 100\% for the first 3 degrees and better than 0\% accuracy for 4-6.

Reviewer: By accuracy you mean recall? "better than 0\%", it's almost 100 for 4-6, and better then 0 for 7-9.

### Accuracy check via simulation: KING workflow

**Description:**

> We are checking accuracy of IBIS+KING workflow with `simulate` command and option `--flow ibis_king`. 

Reviewer: ```suggestion
We are checking accuracy of IBIS+KING workflow with `simulate` command and option `--flow ibis-king`. 
```

> It will take roughly an hour. 

Reviewer: ```suggestion
It takes roughly an hour. 
```

**Command:**

```
docker run --rm --privileged -it -v /media/data1/relatives:/media -v /etc/localtime:/etc/localtime:ro genx_relatives:latest launcher.py simulate --ref-directory /media/big_ref --cores 8 --directory /media/runs/sim-ibis-king-small --flow ibis_king --assembly hg37 --real-run
```

**Desired result:**

> Accuracy of almost 100\% for the first 3 degrees and better than 0\% accuracy for 4-6.

Reviewer: See comment above.

We perform preprocessing and search for SNPs. 

**Command:**

```

# download data
wget https://evolbio.ut.ee/khazar/new_data_in_paper.{bed,bim,fam}

# replace _ in ids with -, because otherwise plink fails in our preprocessing 
sed -i 's/_/-/g' new_data_in_paper.fam

# convert data to vcf.gz
plink --bfile new_data_in_paper --recode vcf-iid bgz --out khazar314 

# run preprocessing
docker run --rm --privileged -it -v /media/data1/relatives:/media -v /etc/localtime:/etc/localtime:ro genx_relatives:latest launcher.py preprocess --ref-directory /media/big_ref --cores 8 --directory /media/runs/khazar --vcf-file /media/datasets/khazar/khazar314.vcf.gz --assembly hg37 --real-run

# run relationship inference
docker run --rm --privileged -it -v /media/data1/relatives:/media -v /etc/localtime:/etc/localtime:ro genx_relatives:latest launcher.py find --ref-directory /media/big_ref --cores 8 --directory /media/runs/khazar --flow ibis --real-run
```

**Desired result:**

`result.csv` file with approximately 56 relatives.

### Performance test on 10K AADR dataset of IBIS workflow

**Description:**

> We need to be sure that GRAPE is working well with big datasets. We chose 10K dataset, because testing on 100K will take too long.

Reviewer: ```suggestion
We need to be sure that GRAPE is working well with big datasets. We chose 10K dataset, because testing on 100K takes too long.
```
